Sage-Bionetworks/BridgePF
# health data v2

New version of health data. Key changes:
- Wiped out all the old health data APIs. Since no one is using health data v1, and since we aren't supporting health data v1, this change will make it so health data v1 API calls will return 400 Bad Request.
- Rewrote the health data controllers, services, and daos. Two APIs:
  *\* createRecord - used by worker apps to create health data records from unpacked uploads
  *\* getRecordsByUploadDate - used by worker apps for exporting health data
- A smattering of DDB marshallers and Jackson serializers for Joda classes.
- A DynamoIndexHelper class, which allows querying against Global Secondary Indices.

Testing done:
- unit tests
- manually tested health data by creating 2 records, then calling getByUploadDate

Next steps:
- add validation in createRecord() to cross-check health codes and schema IDs
- remove the deprecated Tracker APIs
- implement worker accounts, so that worker accounts don't need to be admins
- make upload validation call createRecord
